src/generator/operations.ts

```diff
@@ -332,14 +325,21 @@ function generateOperation(clientName: string, op: Operation, imports: ImportMan
     text += '\t}\n';
     text += `\tpoller := &${camelCase(op.language.go!.pollerType.name)}{\n`;
     text += '\t\t\tpt: pt,\n';
+    if (isPageableOperation(op)) {
+      text += `\t\t\trespHandler: client.${camelCase(op.language.go!.pageableType.name)}HandleResponse,\n`;
+    }
     text += '\t\t\tpipeline: client.p,\n';
     text += '\t}\n';
     text += '\tresult.Poller = poller\n';
     // http pollers will simply return an *http.Response
     if (op.language.go!.pollerType.name === 'HTTPPoller') {
       text += '\tresult.PollUntilDone = func(ctx context.Context, frequency time.Duration) (*http.Response, error) {\n';
     } else {
-      text += `\tresult.PollUntilDone = func(ctx context.Context, frequency time.Duration)(*${(<SchemaResponse>op.responses![0]).schema.language.go!.responseType.name}, error) {\n`;
+      if (isPageableOperation(op)) {
```

Comment:
This could probably be refactored a bit to avoid the duplication.

Reply:
Done